messaging/broker_test.go

```diff
@@ -16,7 +17,7 @@ import (
 // Ensure that opening a broker without a path returns an error.
 func TestBroker_Open_ErrPathRequired(t *testing.T) {
 	b := messaging.NewBroker()
-	if err := b.Open("", "http://127.0.0.1:8080"); err != messaging.ErrPathRequired {
+	if err := b.Open("", &url.URL{Host: "127.0.0.1:8080"}); err != messaging.ErrPathRequired {
```

Comment:
I really like this move to a typed URL. Strings are fine, but it's nice to move away from them as soon as possible.